plumbing/object/commitnode.go

```diff
@@ -0,0 +1,310 @@
+package object
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/commitgraph"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+)
+
+// CommitNode is generic interface encapsulating either Commit object or
+// graphCommitNode object
+type CommitNode interface {
+	ID() plumbing.Hash
+	Tree() (*Tree, error)
+	CommitTime() time.Time
+}
+
+// CommitNodeIndex is generic interface encapsulating an index of CommitNode objects
+// and accessor methods for walking it as a directed graph
+type CommitNodeIndex interface {
+	NumParents(node CommitNode) int
+	ParentNodes(node CommitNode) CommitNodeIter
+	ParentNode(node CommitNode, i int) (CommitNode, error)
+	ParentHashes(node CommitNode) []plumbing.Hash

```

Comment:
Looks like I can move this to `CommitNode` so it looks more like the current `Commit` API.